shell/platform/android/io/flutter/embedding/engine/FlutterEnginePluginRegistry.java

```diff
@@ -294,10 +300,37 @@ public void attachToActivity(@NonNull Activity activity, @NonNull Lifecycle life
             + activity
             + "."
             + (isWaitingForActivityReattachment ? " This is after a config change." : ""));
-    // If we were already attached to an Android component, detach from it.
-    detachFromAndroidComponent();
+    if (this.exclusiveActivity != null) {
+      this.exclusiveActivity.detachFromFlutterEngine();
```

Comment:
Ya, that's the last piece I noted in the PR summary above. I think the place to put it is right (e.g. consolidating the place where we attach/detach, (e.g. if a person makes a custom activity, there should be burden to attach it to the engine in just one API instead of 3 or 4)). 

The name is not right. The easiest solution is to just name this FlutterEngineConnectionRegistry since this file here is sibling here with the engine rather than in the plugins package. For super maintenance, we can also split this into 2 files to make it shorter than 800 LOC. But I don't think we need to do that right now.

Reply:
I've renamed the package private class FlutterEngineConnectionRegistry to reflect its role.